src/main/java/org/reaktivity/nukleus/tcp/internal/writer/stream/StreamFactory.java

```diff
@@ -186,10 +207,33 @@ private void doCleanup()
 
         private int handleWrite()
         {
-            // TODO: partial write completed
+            ByteBuffer writeBuffer = writeSlab.get(slot);
+            if (writeBuffer == null)
+            {
+                return 0;
+            }
+
+            int bytesWritten = 0;
+            try
+            {
+                bytesWritten = channel.write(writeBuffer);
```

Comment:
Given the desire to free up the `Slab` as soon as possible, would it make sense to spin write here also?